Update samples to new csproj files

nicolaspierre1990 opened the pull request.

Commit:
update to new csproj system, clean-up code base, update dependencies

Commit:
add readme to folder structure

Commit:
more cleaning

Commit:
yupdate

drewnoakes commented on src/Beacon/BeaconDemo/App.config:
```diff
@@ -1,6 +1,6 @@
-﻿<?xml version="1.0" encoding="utf-8" ?>
+<?xml version="1.0" encoding="utf-8"?>
```
All these `App.config` files can be deleted.

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,29 @@
-    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net48</TargetFramework>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
```
I don't think we need binding redirects. Perhaps remove these.

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,29 @@
-    <AssemblyName>BeaconDemo</AssemblyName>
-    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net48</TargetFramework>
```
Perhaps move the target framework to a `Directory.Build.props` file in the `src` folder.

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,29 @@
+    <TargetFramework>net48</TargetFramework>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <AssemblyTitle>BeaconDemo</AssemblyTitle>
+    <Product>BeaconDemo</Product>
```
These are redundant, as they default to the name of the project file (which is identical).

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,29 @@
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <AssemblyTitle>BeaconDemo</AssemblyTitle>
+    <Product>BeaconDemo</Product>
+    <Copyright>Copyright ©  2016</Copyright>
```
I would remove these `Copyright` properties, personally (unless they have real values in them).

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,29 @@
+    <EmbeddedResource Remove="Backup1\**" />
+    <EmbeddedResource Remove="Backup\**" />
+    <None Remove="Backup1\**" />
+    <None Remove="Backup\**" />
```
These `Remove`d items are not needed.

drewnoakes commented on src/Beacon/BeaconDemo/BeaconDemo.csproj:
```diff
@@ -1,113 +1,29 @@
   <ItemGroup>
-    <None Include="App.config" />
-    <None Include="packages.config" />
+    <PackageReference Include="AsyncIO" Version="0.1.69" />
+    <PackageReference Include="NaCl.Net" Version="0.1.13" />
+    <PackageReference Include="NetMQ" Version="4.0.1.8" />
+    <PackageReference Include="System.Memory" Version="4.5.5" />
+    <PackageReference Include="System.Numerics.Vectors" Version="4.5.0" />
+    <PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.4" />
```
It would be handy to specify package versions centrally. This can be done in a `Directory.Build.targets` file.

In `.csproj` files you'd omit the `Version`. i.e.:

```xml
<PackageReference Include="AsyncIO" />
```

Then in that `.targets` file you'd have:

```xml
<PackageReference Update="AsyncIO" Version="0.1.69" />
...
```

drewnoakes commented on src/Beacon/BeaconDemo/Bus.cs:
```diff
@@ -1,9 +1,9 @@
-﻿using System;
+﻿using NetMQ;
+using NetMQ.Sockets;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Net;
-using NetMQ;
-using NetMQ.Sockets;
```
It's easier to review a huge PR like this if you keep unrelated refactorings separate.

drewnoakes commented on src/Majordomo/MDPBrokerProcess/MDPBrokerProcess.csproj:
```diff
@@ -1,122 +1,33 @@
+    <TargetFramework>net48</TargetFramework>
+    <AssemblyTitle>MDPBrokerProcess</AssemblyTitle>
+    <Product>MDPBrokerProcess</Product>
+    <AssemblyVersion>1.0.15052.1</AssemblyVersion>
+    <FileVersion>1.0.15052.1</FileVersion>
```
I don't think we need versions on these.

drewnoakes commented on src/Majordomo/MDPClientAsyncExample/MDPClientAsyncExample.csproj:
```diff
@@ -1,80 +1,34 @@
+    <AssemblyTitle>MDPClientAsyncExample</AssemblyTitle>
+    <Product>MDPClientAsyncExample</Product>
+    <Copyright>Copyright ©  2016</Copyright>
+    <OutputPath>bin\$(Configuration)\</OutputPath>
```
I wouldn't override `OutputPath` here (or elsewhere) unless there's a good reason to do so. It shouldn't be necessary.

drewnoakes commented on src/Majordomo/MDPClientExample/MDPClientExample.csproj:
```diff
@@ -1,16 +1,7 @@
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net48</TargetFramework>
     <PublishUrl>publish\</PublishUrl>
```
Looks like all properties from `<PublishUrl>` onwards can be removed.

drewnoakes commented on src/Majordomo/MDPClientExample/MDPClientExampleProgram.cs:
```diff
@@ -63,7 +63,7 @@ private static void Main(string[] args)
                 using (var session = new MDPClient("tcp://localhost:5555", id))
                 {
-                    if (verbose)
+                    //if (verbose)
```
Why this change?

nicolaspierre1990 replied:
Probably when trying things, reverted

drewnoakes commented on src/Majordomo/MDPWorkerExample/MDPWorkerExampleProgram.cs:
```diff
@@ -37,7 +37,7 @@ private static void Main (string[] args)
                     session.HeartbeatDelay = TimeSpan.FromMilliseconds (10000);
                     // logging info to be displayed on screen
-                    if (verbose)
+                    //if (verbose)
```
Ditto.

nicolaspierre1990 replied:
See previous

drewnoakes commented on src/Majordomo/MajordomoProtocolTests/MDPBrokerTests.cs:
```diff
@@ -105,7 +105,7 @@ public async void Run_ReceiveREADYMessageFromWorker_LogSuccessfulRegistration ()
                 // check on the logging
                 Assert.That (log.Count, Is.EqualTo (2));
-                Assert.That (log[1], Is.StringContaining ("added to service echo"));
+                Assert.That (log[1], Does.Contain("added to service echo"));
```
Why this change?

nicolaspierre1990 replied:
Removed in Nunit 3.0
https://docs.nunit.org/articles/nunit/release-notes/breaking-changes.html

drewnoakes commented on src/Pirate Pattern/Lazy Pirate/LazyPirate.Client2/Program.cs:
```diff
@@ -44,25 +44,25 @@ private static void SubscriberContinuousLoop(string pubSubAddress, string reques
-                            default:
-                                throw new ArgumentOutOfRangeException();
-                        }
+                        //RequestSocket.ProgressTopic progressTopic;
+                        //Enum.TryParse(topic, out progressTopic);
+                        //switch (progressTopic)
+                        //{
+                        //    case RequestSocket.ProgressTopic.Send:
+                        //        Console.WriteLine("C: Sending {0}", requestString);
+                        //        break;
+                        //    case RequestSocket.ProgressTopic.Retry:
+                        //        Console.WriteLine("C: No response from server, retrying...");
+                        //        break;
+                        //    case RequestSocket.ProgressTopic.Failure:
+                        //        Console.WriteLine("C: Server seems to be offline, abandoning");
+                        //        break;
+                        //    case RequestSocket.ProgressTopic.Success:
+                        //        Console.WriteLine("C: Server replied OK");
+                        //        break;
+                        //    default:
+                        //        throw new ArgumentOutOfRangeException();
+                        //}
```
Why this change?

nicolaspierre1990 replied:
@drewnoakes: Enum was made private, any reason for this ?

drewnoakes commented on src/Samples.sln:
```diff
@@ -1,63 +1,63 @@
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HelloWorld", "HelloWorld\HelloWorld.csproj", "{7AB16EE9-4A79-4BFB-8C0B-D1A076E34896}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "HelloWorld", "HelloWorld\HelloWorld.csproj", "{7AB16EE9-4A79-4BFB-8C0B-D1A076E34896}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WeatherUpdateClient", "WeatherUpdater\WeatherUpdateClient\WeatherUpdateClient.csproj", "{37B320C0-9E44-44EF-B89A-E671DBA4986B}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "WeatherUpdateClient", "WeatherUpdater\WeatherUpdateClient\WeatherUpdateClient.csproj", "{37B320C0-9E44-44EF-B89A-E671DBA4986B}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WeatherUpdateServer", "WeatherUpdater\WeatherUpdateServer\WeatherUpdateServer.csproj", "{5B8F7740-EFD5-46B9-8A96-2E1F47385295}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "WeatherUpdateServer", "WeatherUpdater\WeatherUpdateServer\WeatherUpdateServer.csproj", "{5B8F7740-EFD5-46B9-8A96-2E1F47385295}"
```
These project type GUID changes are not necessary and should be reverted.

drewnoakes commented on src/Titanic Pattern/TitanicUsageExample/TitanicWorkerExample/TitanicWorkerExample.csproj:
```diff
@@ -1,133 +1,39 @@
     <UpdateAssemblyInfoVersion>False</UpdateAssemblyInfoVersion>
-    <AssemblyVersionSettings>
-    </AssemblyVersionSettings>
-    <AssemblyFileVersionSettings>
-    </AssemblyFileVersionSettings>
-    <AssemblyInfoVersionSettings>
-    </AssemblyInfoVersionSettings>
+    <AssemblyVersionSettings />
+    <AssemblyFileVersionSettings />
+    <AssemblyInfoVersionSettings />
+    <AssemblyTitle>MDPWorkerExample</AssemblyTitle>
+    <Product>MDPWorkerExample</Product>
+    <Copyright>Copyright ©  2015</Copyright>
```
All these can be deleted.

drewnoakes commented on src/Beacon/BeaconDemo/Properties/AssemblyInfo.cs:
```diff
@@ -2,35 +2,10 @@
 using System.Runtime.CompilerServices;
```
All these `AssemblyInfo.cs` files can be deleted. These projects aren't using COM.

drewnoakes reviewed (commented):
Nice work. Left some comments on further clean up that would be good to add, to tidying things even further.

I'm curious, did you convert these files by hand, or did you use a tool? If it was a tool, which did you use?

nicolaspierre1990 commented:
I lost track of this, I used project2015to2017.migrate2019.tool for the migration, I will fix the comments

Commit:
clean-up unused stuff

Commit:
clean-up references p1

Commit:
Revert "clean-up references p1"

This reverts commit de969aebecf3b761497886e862218a70c230268a.

Commit:
fixed comments

nicolaspierre1990 commented:
still WIP

nicolaspierre1990 committed:
Added Untitled Diagram.drawio

drewnoakes committed:
Specify package versions centrally

drewnoakes committed:
Add global usings

drewnoakes committed:
Remove redundant package references

drewnoakes committed:
Use file-scoped namespaces

drewnoakes committed:
Use top-level statements for several project entrypoints

drewnoakes committed:
Use simple using statements

drewnoakes committed:
Use "is null" instead of ReferenceEquals

drewnoakes committed:
Inline variable declarations

drewnoakes committed:
Use target-typed new

drewnoakes committed:
Remove drawio document

drewnoakes committed:
Update README

drewnoakes committed:
Get tests running in VS

drewnoakes commented:
@nicolaspierre1990 I pulled your branch locally to play with it, then went through and applied a bunch more modernisations to the code across the solution. Pushed those here, and will merge them in one hit.

drewnoakes committed:
Don't use async void for tests

The test runner cannot know when the test completes if we don't return a task.

drewnoakes committed:
Restore assembly version

The unit tests are validating against this version. It must have been removed during the migration to SDK-style projects.

drewnoakes committed:
Avoid calling Task.Wait where possible

nicolaspierre1990 commented:
Sorry I forgot about this PR :D Glad it's merged
